File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    for size in sizes:
        if size[0] < size[1]:
            tmp = size[0]
            size[0] = size[1]
            size[1] = tmp
    r_max = 0
    c_max = 0
    
    for size in sizes:
        if r_max < size[0]:
            r_max = size[0]
        if c_max< size[1]:
            c_max = size[1]
            
    return r_max * c_max

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        
        for(int i = 0; i < sizes.length; i++) {
            if(sizes[i][0] < sizes[i][1]) {
                int tmp = sizes[i][0];
                sizes[i][0] = sizes[i][1];
                sizes[i][1] = tmp;
            }
        }
        
        int rMax = 0;
        int cMax = 0;
        
        for(int[] size : sizes) {
            if(size[0] > rMax) {
                rMax = size[0];
            }
            if(size[1] > cMax) {
                cMax = size[1];
            }
        }
            
        return rMax * cMax;
    }
}